src/auth.ts

```diff
@@ -102,6 +102,10 @@ export async function auth(
         ...authentication,
       };
 
+      // if (state.onTokenCreated) {
+      state.onTokenCreated(state.authentication, options.type);
```

Comment:
Would this be the only place where we want to invoke the `onTokenCreated` or there are other places?

Reply:
only if `options.type` is `reset`. [`checkToken`](https://github.com/octokit/oauth-methods.js#checktoken) does not create a new token, it only checks if the current one is valid.

We also need to add the hook in `src/get-authentication.ts` at https://github.com/octokit/auth-oauth-user.js/blob/64558b5e44fd235ae40fe9529653da18f776fb60/src/get-authentication.ts#L39 and https://github.com/octokit/auth-oauth-user.js/blob/64558b5e44fd235ae40fe9529653da18f776fb60/src/get-authentication.ts#L59

Reply:
> only if `options.type` is `reset`. [`checkToken`](https://github.com/octokit/oauth-methods.js#checktoken) does not create a new token, it only checks if the current one is valid.

Covered here now: https://github.com/octokit/auth-oauth-user.js/pull/90/files#diff-84a5d130fb4cccb80f78601d140f1d5397dc0272f94cea672a8470539fcf6dc7R109-R114

> We also need to add the hook in `src/get-authentication.ts` at
> 
> https://github.com/octokit/auth-oauth-user.js/blob/64558b5e44fd235ae40fe9529653da18f776fb60/src/get-authentication.ts#L39
> 
> and
> https://github.com/octokit/auth-oauth-user.js/blob/64558b5e44fd235ae40fe9529653da18f776fb60/src/get-authentication.ts#L59

After checking the code, would it be simpler if we just call it at this point, after calling `getAuthentication()` a few lines above?

https://github.com/octokit/auth-oauth-user.js/pull/90/files#diff-84a5d130fb4cccb80f78601d140f1d5397dc0272f94cea672a8470539fcf6dc7R83-R86